### ISSUE TITLE
Document macOS security-related information

### ISSUE THREAD

**Issue by stuartmorgan** [flutter/flutter#52902]
There are things developers will need to know about security settings for macOS Flutter applications; currently (for legacy reasons) that content lives in the FDE repo at https://github.com/google/flutter-desktop-embedding/blob/master/macOS-Security.md. Now that we have `flutter create` support for macOS, it should be on the Flutter site instead.

For now this should probably live on https://flutter.dev/desktop since all the other macOS content is currently there.

There's a lot of content there at the moment; I'm happy to work on drafts with someone if we need to slim it down (e.g., we could potentially remove discussion of hardened runtime and leave that for people to figure out on their own since I don't think disabling the sandbox will be all that common, and there's nothing really Flutter-specific in that section).

/cc @csells to help route.

**Comment by csells** [comment]
@legalcodes can you help us here?

**Comment by legalcodes** [comment]
Yep, happy to help here! @stuartmorgan let me know how you'd like to proceed.

> I'm happy to work on drafts with someone if we need to slim it down 

On the other hand, if we'd like to keep this content (I think it's really helpful) I'm sure we can find a way to do so -- we could give it its own page, for example.

**Comment by stuartmorgan** [comment]
@csells Do we still want all macOS documentation on the one desktop page, or should this be a separate page?

**Comment by sfshaza2** [comment]
@legalcodes is no longer on our team. So you'll be dealing with me.  :)

**Comment by sfshaza2** [comment]
So, @csells and @stuartmorgan, signing apps is a deployment issue, yes? I propose that I create a Deployment page for desktop. For now, it would only discuss macOS, but later would also cover Windows and Linux. I don't want to keep jamming all desktop info into the /desktop page.

**Comment by stuartmorgan** [comment]
> So, @csells and @stuartmorgan, signing apps is a deployment issue, yes?

Well, yes and no. Signing itself only matters for deployment, but the entitlement discussions there apply to development as well. Notably, if you make any internet requests you need an entitlement, and people will run into (and already have been running into) that during development.

Comparing with existing platforms:
- iOS doesn't have such an entitlement
- Android does, but it's a combined incoming/outgoing entitlement, so it's on for debug in the Flutter app template to make the observatory work; as a result it only matters for release builds, and is accordingly covered in the deployment section. On macOS incoming and outgoing are separate, and only incoming is on in debug in the template, so if developers want to make any outgoing requests they need to add it.

So there's not a good precedent for where else to put it, but I don't think Deployment is the right place for that part of it.

**Comment by sfshaza2** [comment]
I see, @stuartmorgan. Chris has scheduled us to talk on Monday, so we can hash it out then.
